components/raftstore/src/store/peer.rs

```diff
@@ -4560,10 +4622,23 @@ where
     ///
     /// 1. pre_transfer_leader on leader:
     ///     Leader will send a MsgTransferLeader to follower.
-    /// 2. execute_transfer_leader on follower
-    ///     If follower passes all necessary checks, it will reply an
-    ///     ACK with type MsgTransferLeader and its promised persistent index.
-    /// 3. ready_to_transfer_leader on leader:
+    /// 2. pre_ack_transfer_leader_msg on follower:
+    ///     If follower passes all necessary checks, it will try to warmup
+    ///     the entry cache.
+    /// 3. ack_transfer_leader_msg on follower:
+    ///     When the entry cache has been warmed up or the operator is timeout,
+    ///     the follower reply an ACK with type MsgTransferLeader and
+    ///     its promised persistent index.
+    ///
+    /// (Condition: when there are remaining pessimistic locks to propose)
```

Comment:
A little confusing,
```suggestion
    /// Special notes for behavior when there are remaining pessimistic locks to propose
    ///    /*explain where and how do we detect this condition*/
```
And isn't `ready_to_transfer_leader` called after `ack_transfer_leader_msg`? Why explain it first?

Reply:
`ready_to_transfer_leader` behaves differently in different context (conditions).